File: erpnext_retail/erpnext_retail/web_form/customer_and_shipping_details/customer_and_shipping_details.js
frappe.ready(function() {
    // bind events here
    frappe.web_form.after_load = () => {

        frappe.web_form.on('customer', () => {
            frappe.call({
                method: 'erpnext_retail.erpnext_retail.web_form.customer_and_shipping_details.customer_and_shipping_details.get_customer',
                args: {
                    customer: frappe.web_form.get_value("customer")
                },
                callback: function(r) {
                    frappe.web_form.set_value("address", r.message[0])
                    frappe.web_form.set_value("second_address", r.message[1])
                    frappe.web_form.set_value("third_address", r.message[2])
                    frappe.web_form.set_value("fourth_address", r.message[3])
                    frappe.web_form.set_value("fifth_address", r.message[4])
                        // console.log(r)
                }
            });

        })

        frappe.web_form.add_button_to_footer("Create Address", "submit", () => {
            frappe.call({
                method: 'erpnext_retail.erpnext_retail.web_form.customer_and_shipping_details.customer_and_shipping_details.create_new_address',
                args: {
                    customer: frappe.web_form.get_value("customer"),
                    new_address: frappe.web_form.get_value("new_address"),
                    city: frappe.web_form.get_value("city"),
                    address_title: frappe.web_form.get_value("address_title"),
                    state: frappe.web_form.get_value("state"),
                    pin_code: frappe.web_form.get_value("pin_code")

                },
                // callback: function(r) {

                //         // console.log(r)
                // }
            });
        })

    };
})

File: erpnext_retail/erpnext_retail/web_form/customer_and_shipping_details/customer_and_shipping_details.py
from __future__ import unicode_literals

import frappe

def get_context(context):
    # do your magic here
    pass


@frappe.whitelist()		
def get_customer(customer):
    address = frappe.db.sql("select a.address_line1,a.address_line2,a.city,a.state,a.pincode from `tabAddress` a LEFT JOIN `tabDynamic Link` dl on dl.parent = a.name LEFT JOIN `tabCustomer` c on dl.link_name = c.name where dl.link_name=%s", customer)
    return address

@frappe.whitelist()		
def create_new_address(customer,new_address,city,address_title,state,pin_code):
    if customer:
        address = frappe.new_doc('Address')
        address.address_title = address_title + customer
        address.address_line1 = new_address
        address.city = city
        address.state = state
        address.pin_code = pin_code
        address.append("links",{
                            'link_doctype':"Customer",
                            'link_name':customer,
                            'link_title':customer                                                       
                        })
            
        address.insert()
        address.save()
        frappe.msgprint("Add New Address Create")
    else:
        frappe.msgprint("Customer Not Null")
        